proposals/0759-react-native-frameworks.md

```diff
@@ -0,0 +1,452 @@
+
+### Benefits
+
+Frameworks that are adhering to requirements described below:
+
+- Will be spotlighted in the Getting Started page of the React Native website (our highest traffic page). Specifically, we’ll recommend to users to use one of the Frameworks which satisfies the requirements.
+- Will be mentioned during public comms as one of the recommended frameworks by the React Native core team.
+- The company/entity behind the Framework will automatically be granted a [partner status](https://github.com/facebook/react-native/blob/main/ECOSYSTEM.md#partners).
+- PRs and Issues coming from the Framework developers will receive higher priorities.
+
+### Requirements
+
+Here is a short summary of what are the requirements for React Native Frameworks.
+
+#### DOs of Frameworks
+
+- **Focus on developer & user experience** - We believe Frameworks should aim to provide a top-notch developer experience. Frameworks that fail to provide good documentation or developer workflows are not likely high quality frameworks, and our users will most likely not want to use them. Similarly, we want to promote frameworks that provide a top-notch user experience and do not ignore basic user needs, such as accessibility, performance, and delightful UIs.
+- **Focus on building cohesive experiences** - Users are looking into deploying universal apps across multiple targets. We’re delegating this responsibility to frameworks that can provide better experiences to users and craft cohesive developer experiences between mobile, web, desktop and beyond.
+- **Focus on abstracting complexity away** - Any initiative that simplifies the developer workflow is welcomed in the framework domain.
+- **Be popular** - We’ll only be recommending solutions that are widely known and adopted by the community.
+- **Open source** - Recommended frameworks must be free, open source, projects. We allow for companies to provide paid services on top of the Open Source framework, but these services should not be a requirement to use the framework and should not have vendor lock-in.
+
+#### DON’Ts of Frameworks
+
+- **Do not hard fork react-native** - As a rule of thumb, you should not GitHub-fork react-native and re-publish it. If you’re doing so, it’s probably due to a lack of APIs that React Native should provide. Please reach out to us and attempt to de-fork your framework.
+- **Do not soft fork the core APIs** - We want to make sure developers need to learn only once how to write a native component. Building abstraction and wrappers is accepted, forking is not, as it makes it harder for users to share components/modules between targets.
+- **Do not fork the ecosystem** - It’s crucial that `react-native-*` libraries can be easily shared and ported to other targets or frameworks (with little to no work). Frameworks should aim at building libraries that work across the whole ecosystem and not solutions that silo the users away.
```

Comment:
Again, I'm having trouble understanding why "forking" is the focus here, especially for a library. Like if I'm using the third-party `react-native-foo` library, I need to add some functionality but the maintainer is not responsive, I would expect I could fork and add what I'd like to my version. What's the problem there?

Reply:
> Like if I'm using the third-party react-native-foo library, I need to add some functionality but the maintainer is not responsive, I would expect I could fork and add what I'd like to my version. What's the problem there?

Forking an unmaintaied library is actually encouraged, as you're providing a solution to one of the biggest pain point of React Native ([source](https://results.stateofreactnative.com/opinions/opinions_pain_points_multiple)).

The sentence here refers to "forking the ecosystem". We want to avoid situations where a framework `foo-` forks a core API and starts encouraging developers do develop libraries on top of that forked API. So you'll end up having say a `foo-push-notification` libraries, rather than a `react-native-push-notification` camera that can easily be used between different frameworks.